webui/react/src/services/decoder.ts

```diff
@@ -464,25 +466,78 @@ export const mapSearchExperiment = (
   };
 };
 
-export const mapV1Experiment = (
+export function mapV1Experiment(
   data: Sdk.V1Experiment,
   jobSummary?: types.JobSummary,
-): types.ExperimentItem => {
+  config?: undefined,
+): types.BulkExperimentItem;
+export function mapV1Experiment(
+  data: Sdk.V1Experiment,
+  jobSummary?: types.JobSummary,
+  config?: Sdk.V1GetExperimentResponse['config'],
+): types.FullExperimentItem;
+export function mapV1Experiment(
+  data: Sdk.V1Experiment,
+  jobSummary?: types.JobSummary,
+  config?: Sdk.V1GetExperimentResponse['config'],
+): types.FullExperimentItem | types.BulkExperimentItem {
+  const continueFn = (value: unknown) => !(value as types.HyperparameterBase).type;
+
+  if (config === undefined) {
+    const hyperparameters = flattenObject<types.HyperparameterBase>(data?.hyperparameters ?? {}, {
+      continueFn,
+    }) as types.HyperparametersFlattened;
+
+    const bulkExpItem: types.BulkExperimentItem = {
+      archived: data.archived,
+      checkpoints: data.checkpointCount,
+      checkpointSize: parseInt(data?.checkpointSize || '0'),
+      description: data.description,
+      duration: data.duration,
+      endTime: data.endTime as unknown as string,
+      externalExperimentId: data.externalExperimentId,
+      externalTrialId: data.externalTrialId,
+      forkedFrom: data.forkedFrom,
+      hyperparameters,
+      id: data.id,
+      jobId: data.jobId,
+      jobSummary: jobSummary,
+      labels: data.labels || [],
+      modelDefinitionSize: data.modelDefinitionSize,
+      name: data.name,
+      notes: data.notes,
+      numTrials: data.numTrials || 0,
+      progress: data.progress != null ? data.progress : undefined,
+      projectId: data.projectId,
+      projectName: data.projectName,
+      resourcePool: data.resourcePool || '',
+      searcherMetricValue: data.bestTrialSearcherMetric,
+      searcherType: data.searcherType,
+      startTime: data.startTime as unknown as string,
+      state: decodeExperimentState(data.state),
+      trialIds: data.trialIds || [],
+      unmanaged: data.unmanaged,
+      userId: data.userId ?? 0,
+      workspaceId: data.workspaceId,
+      workspaceName: data.workspaceName,
+    };
+    return bulkExpItem;
+  }
```

Comment:
might make sense to DRY this up -- remember that a bulk experiment is a subset of a full experiment, so you could start with the bulk experiment, and if config is undefined, return it otherwise add the config and configraw properties.

Reply:
fixed